test/map/transform.test.cpp

```diff
@@ -514,3 +520,62 @@ TEST(Transform, DefaultTransform) {
     ASSERT_FALSE(transform.resize({{ max, max }}));
     testConversions(nullIsland, center);
 }
+
+TEST(Transform, TileCoordinate)
```

Comment:
Let's put all the `TileCoordinate`-related tests -- the ones here and in tile_point.test.cpp -- in tile_coordinate.test.cpp.

I've been campaigning to get rid of `TileCoordinate` for awhile ([1](https://github.com/mapbox/mapbox-gl-native/pull/3873#issuecomment-182679439), [2](https://github.com/mapbox/mapbox-gl-native/issues/3785), [3](https://github.com/mapbox/mapbox-gl-native/issues/3529#issuecomment-179493928)) -- keeping it and its tests each confined to a single file at least limits the spread as much as possible.